src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/events/EnvelopeHandler.java

```diff
@@ -60,15 +68,45 @@ public EnvelopeProcessingResult handleEnvelope(Envelope envelope) {
                 }
             case SUPPLEMENTARY_EVIDENCE_WITH_OCR:
             case EXCEPTION:
-            case NEW_APPLICATION:
                 return new EnvelopeProcessingResult(createExceptionRecord(envelope), EXCEPTION_RECORD);
+            case NEW_APPLICATION:
+                return processNewApplication(envelope, deliveryCount);
             default:
                 throw new UnknownClassificationException(
                     "Cannot determine CCD action for envelope - unknown classification: " + envelope.classification
                 );
         }
     }
 
+    private EnvelopeProcessingResult processNewApplication(Envelope envelope, long deliveryCount) {
+        var caseCreationResult = caseCreator.createCase(envelope);
+
+        switch (caseCreationResult.resultType) {
+            case CASE_CREATED:
+            case CASE_ALREADY_EXISTS:
+                paymentsProcessor.createPayments(envelope, caseCreationResult.caseCcdId, false);
+                return new EnvelopeProcessingResult(caseCreationResult.caseCcdId, CASE_CREATED);
+            case ABORTED_WITHOUT_FAILURE:
+            case UNRECOVERABLE_FAILURE:
+                return new EnvelopeProcessingResult(createExceptionRecord(envelope), EXCEPTION_RECORD);
+            case POTENTIALLY_RECOVERABLE_FAILURE:
+                if (deliveryCount < MAX_RETRIES_FOR_POTENTIALLY_RECOVERABLE_FAILURES) {
+                    // let the application retry later
+                    throw new CaseCreationException("Case creation failed due to a potentially recoverable error");
```

Comment:
Currently, the only way of preventing `MessageProcessor` from completing the queue message is by throwing an exception.